buildpipeline/perf-pipeline.groovy

```diff
@@ -79,41 +62,21 @@ def windowsPerf(String arch, String config, String uploadString, String runType,
         String runXUnitCommonArgs = "-arch ${arch} -configuration ${config} -generateBenchviewData \"%WORKSPACE%\\Microsoft.Benchview.JSONFormat\\tools\" ${uploadString} ${pgoTestFlag} -runtype ${runType} ${testEnv} -optLevel ${opt_level} -jitName ${jit} -outputdir \"%WORKSPACE%\\bin\\sandbox_logs\""
         if (scenario == 'perf') {
             String runXUnitPerfCommonArgs = "${runXUnitCommonArgs} -stabilityPrefix \"START \"CORECLR_PERF_RUN\" /B /WAIT /HIGH /AFFINITY 0x2\""
-            if (test == 'first half' || testFolder == 'all')
+            if (slice == 4 || slice == -1)
```

Comment:
It seems odd to have 4 be a special value and make PerfLab and PerfLab a special case. Would it be reasonable to instead roo the paths in the JSON file from the coreclr/test folder and include this one there? I'm guessing the other reason this is here is because it needs the -library flag to run. Could you include the flag as an optional "additional args" field within the json?

Reply:
We should remove slice specific knowledge from the groovy file.  The way I think about it is that the goal should be to be able to modify the number of slices with a one line change to the groovy file - just changing the number of jobs that get created.  Everything else should be encapsulated into the configuration or scripting which is stored in GitHub and can be used when each slice actually runs.

If in fact slice 4 requires a different run command, then we should encapsulate that into the config in some way.

Reply:
What it is is that perflab is a special case because it is a) in a different directory (not CodeQuality) and b) it is a library and requires the -library flag. I put it in slice 4 because of load balancing, and this code is still here because I was trying to simplify any additional scripting I had to do (ie, not having to have the python script I wrote have any knowledge about run-xunit-perf.cmd/sh).